src/ol_infrastructure/infrastructure/mongodb/__main__.py

```diff
@@ -0,0 +1,209 @@
+import json
+from itertools import chain
+
+from pulumi import Config, ResourceOptions
+from pulumi.stack_reference import StackReference
+from pulumi_aws import ec2, iam
+
+from ol_infrastructure.lib.aws.ec2_helper import (
+    InstanceTypes,
+    build_userdata,
+    debian_10_ami,
+    default_egress_args,
+)
+from ol_infrastructure.lib.aws.iam_helper import lint_iam_policy
+from ol_infrastructure.lib.ol_types import AWSBase
+from ol_infrastructure.lib.pulumi_helper import parse_stack
+from ol_infrastructure.providers.salt.minion import (
+    OLSaltStackMinion,
+    OLSaltStackMinionInputs,
+)
+
+stack_info = parse_stack()
+env_config = Config("environment")
+mongodb_config = Config("mongodb")
+salt_config = Config("saltstack")
+environment_name = f"{stack_info.env_prefix}-{stack_info.env_suffix}"
+business_unit = env_config.get("business_unit") or "operations"
+aws_config = AWSBase(tags={"OU": business_unit, "Environment": environment_name})
+network_stack = StackReference(f"infrastructure.aws.network.{stack_info.name}")
+operations_vpc = network_stack.require_output("operations_vpc")
+consul_stack = StackReference(
+    "infrastructure.consul."  # noqa: WPS237, WPS221
+    f"{stack_info.namespace.rsplit('.', 1)[1]}."  # noqa: WPS237, WPS221
+    f"{stack_info.name}"
+)
+destination_vpc = network_stack.require_output(env_config.require("vpc_reference"))
+
+mongodb_instance_policy = {
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Action": [
+                "ec2:DescribeInstances",
+                "ec2:DescribeAvailabilityZones",
+                "ec2:DescribeRegions",
+                "ec2:DescribeSecurityGroups",
+                "ec2:DescribeTags",
```

Comment:
Is Mongo using some form of ec2 autodiscovery for clustering? Or is this copy :spaghetti: from the ES setup?